genome/bin_statistic.py

```
@@ -33,36 +38,169 @@ def contig2bin(outdir: PathLike, contig2bin_tsv: PathLike, contigs: PathLike):
+            b: {} for b in self.contig2bin_tsv["bin"].unique()
+        }
+        for i in self.contigs:
+            if i.name in self.contig2bin_tsv.index:
+                bin2seqs[str(self.contig2bin_tsv.loc[i.name, "bin"])][i.name] = i
+        return bin2seqs
+
+    def parse_contigs(self, contigs: PathLike | Iterable[SeqRecord.SeqRecord]):
+        if isinstance(contigs, str) or isinstance(contigs, Path):
+            try:
+                return lambda: SeqIO.parse(contigs, "fasta")
+            except ValueError as e:
```

Ruff (0.11.9)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)